src/main/java/net/minecraftforge/event/world/WorldEvent.java

```diff
@@ -118,4 +126,83 @@ public ServerLevelData getSettings()
             return settings;
         }
     }
+
+     /**
+     * Fired when building a list of all possible entities that can spawn at the specified location.
+     *
+     * <p>If an entry is added to the list, it needs to be a globally unique instance.</p>
+     *
+     * The event is called in {@link net.minecraft.world.level.NaturalSpawner#mobsAt(ServerLevel,
+     * StructureManager, ChunkGenerator, MobCategory, RandomSource, BlockPos)}.</p>
+     *
+     * <p>This event is {@linkplain Cancelable cancellable}, and does not {@linkplain HasResult have a result}.
+     * Canceling the event will result in an empty list, meaning no entity will be spawned.</p>
+     */
+    @Cancelable
+    public static class PotentialSpawns extends WorldEvent
+    {
+        private final MobCategory mobcategory;
+        private final BlockPos pos;
+        private final List<MobSpawnSettings.SpawnerData> list;
+        private final List<MobSpawnSettings.SpawnerData> view;
+
+        public PotentialSpawns(LevelAccessor level, MobCategory category, BlockPos pos, WeightedRandomList<MobSpawnSettings.SpawnerData> oldList)
+        {
+            super(level);
+            this.pos = pos;
+            this.mobcategory = category;
+            if (!oldList.isEmpty())
+                this.list = new ArrayList<>(oldList.unwrap());
+            else
+                this.list = new ArrayList<>();
+
+            this.view = Collections.unmodifiableList(list);
+        }
+
+	/**
```

Comment:
Indentation errors: the beginning of the javadoc comments and the method declarations are using tabs rather than spaces.